pxr/usdImaging/lib/usdviewq/appController.py

```diff
@@ -3186,31 +3186,35 @@ def _showPrimContextMenu(self, item):
         self.contextMenu = PrimContextMenu(self._mainWindow, item, self)
         self.contextMenu.exec_(QtGui.QCursor.pos())
 
-    def setFrame(self, frameIndex, forceUpdate=False):
-        frameAtStart = self._dataModel.currentFrame
+    def setFrame(self, frame):
+        """Set the `frame`.
+
+        Args:
+            frame (str|int|float): The new frame value.
+        """
+        frameIndex = self._findClosestFrameIndex(float(frame))
+        self._setFrameIndex(frameIndex)
+
+    def _setFrameIndex(self, frameIndex):
+        """Set the `frameIndex`.
 
-        frame = self._timeSamples[int(frameIndex)]
-        if self._dataModel.currentFrame.GetValue() != frame:
-            minDist = 1.0e30
-            closestFrame = None
-            for t in self._timeSamples:
-                dist = abs(t - frame)
-                if dist < minDist:
-                    minDist = dist
-                    closestFrame = t
+        Args:
+            frameIndex (int): The new frame index value.
+        """
+        # Ensure the frameIndex exists, if not, return.
+        try:
+            frame = self._timeSamples[frameIndex]
+        except IndexError:
+            return
 
-            if closestFrame is None:
-                return
+        frameIndex = int(frameIndex)
 
-            self._dataModel.currentFrame = Usd.TimeCode(closestFrame)
+        if self._dataModel.currentFrame != frame:
+            self._dataModel.currentFrame = Usd.TimeCode(frame)
```

Comment:
Just to ensure that the comparison is happening inthe right space, should we convert `frame` to a TimeCode *before* doing the comparison?